### ISSUE TITLE
Unable to mock classes that make call to 'object.hasOwnProperty'

### ISSUE THREAD

**Issue by manda-linda** [NagRock/ts-mockito#28]
Calling mock(TestService) results in the error 'TypeError: methodStub.hasMatchingInAnyGroup is not a function'

**Comment by NagRock** [comment]
Can you provide some more code?

**Comment by NagRock** [comment]
Or even better github repo example?

**Comment by manda-linda** [comment]
Here is an example of a class that has this error when calling mock(EventsManager)

```
class EventsManager {
    private eventMap;

    constructor() {
        this.eventMap = {};
    }
    public subscribe(name: string, func: (...args: any[]) => void): (() => void) {
        if (!this.eventMap.hasOwnProperty(name)) {
            this.eventMap[name] = [];
        }
        return () => {
            delete this.eventMap[name][currentIndex];
        };
    };

    public publish(name: string, data?: any, runNow?: boolean): void {
        if (this.eventMap.hasOwnProperty(name)) {
            // do something
        }
    };
}
```

**Comment by NagRock** [comment]
Thanks. Will take a look at this.

**Comment by NagRock** [comment]
Should be fixed in 2.0.2 version. @manda-linda once again thanks for reporting this issue!
Please reopen if it still occurs.
